File: Data Structures & Algorithms/eating-bananas/submission-0.js
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles, h) {
        let l = 1, r = Math.max(...piles)
        let res = Infinity

        while(l <= r) {
            const k = Math.floor((l + r) / 2)

            let totalTime = 0
            for(let pile of piles) {
                let currentTime = Math.ceil(pile / k)
                totalTime += currentTime
            }

            console.log(totalTime)

            if(totalTime > h) {
                l = k + 1
            } else if(totalTime <= h) {
                r = k - 1
                res = Math.min(res, k)
            }
        }

        return res

    }
}

File: Data Structures & Algorithms/minimum-difference-between-highest-and-lowest-of-k-scores/submission-0.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums = sorted(nums)
        res = math.inf
        for i in range(0, len(nums)):
            if i + k - 1 < len(nums):
                diff = nums[i + k - 1] - nums[i]
                res = min(res, diff)
                
        return int(res)